xarray/test/test_dataset.py

```diff
@@ -1849,6 +1849,75 @@ def test_convert_dataframe_with_many_types_and_multiindex(self):
             expected = expected.drop(['h'], axis=1)
         assert roundtripped.equals(expected)
 
+    def test_to_and_from_dict(self):
+        """
+        <xarray.Dataset>
+        Dimensions:  (t: 10)
+        Coordinates:
+          * t        (t) <U1 'a' 'b' 'c' 'd' 'e' 'f' 'g' 'h' 'i' 'j'
+        Data variables:
+            a        (t) float64 0.6916 -1.056 -1.163 0.9792 -0.7865 0.7831 0.4174 ...
+            b        (t) float64 1.32 0.1954 1.91 1.39 0.519 -0.2772 -1.316 1.111 ...
+        """
+        x = np.random.randn(10)
+        y = np.random.randn(10)
+        t = list('abcdefghij')
+        ds = Dataset(OrderedDict([('a', ('t', x)),
+                                  ('b', ('t', y)),
+                                  ('t', ('t', t))]))
+        expected = {
+                    'coords': {
+                               't':{
+                                    'dims': ['t'],
+                                    'data': t,
+                                    'attrs': {}
+                                    }
+                              },
+                    'attrs': {}, 
+                    'dims': ['t'],
```

Comment:
What is the value of that? For `DataArray` it will have to be a list or an `OrderedDict` anyways so that the shape of the data matches the shape of the dims.